lib/stdlib/src/ets.erl

```diff
@@ -478,6 +479,21 @@ slot(_, _) ->
 take(_, _) ->
     erlang:nif_error(undef).
 
+-spec take_all_objects(Table) -> [Object] when
+      Table :: table(),
+      Object :: tuple().
+
+take_all_objects(T) ->
+    ets:safe_fixtable(T, true),
+    Objects = internal_take_all(T, ets:first(T), []),
+    ets:safe_fixtable(T, false),
+    Objects.
+
+internal_take_all(_, '$end_of_table', Acc) ->
+    Acc;
+internal_take_all(T, Key, Acc) ->
+    ets:internal_take_all(T, ets:next(T, Key), ets:take(T, Key) ++ Acc).
```

Comment:
According to my benchmarks, appending with `++` is the same as flattening afterward. Optimizing the standard case with a single element, like in `set`, makes this about three times faster. Here's the benchmark results:

```
Operating System: macOS
CPU Information: Apple M1 Pro
Number of Available Cores: 10
Available memory: 32 GB
Elixir 1.14.2
Erlang 25.1

Benchmark suite executing with the following configuration:
warmup: 2 s
time: 5 s
memory time: 0 ns
reduction time: 0 ns
parallel: 1
inputs: none specified
Estimated total run time: 21 s

Benchmarking Append ...
Benchmarking Flatten ...
Benchmarking Optimized ...

Name                ips        average  deviation         median         99th %
Optimized       32.73 K       30.55 μs   ±166.00%       32.67 μs       46.29 μs
Append          10.72 K       93.25 μs     ±3.64%       92.33 μs      106.25 μs
Flatten         10.70 K       93.47 μs     ±4.88%       92.33 μs      106.97 μs

Comparison:
Optimized       32.73 K
Append          10.72 K - 3.05x slower +62.70 μs
Flatten         10.70 K - 3.06x slower +62.92 μs
```

Then again, there may be a faster way to accomplish all of this 🤷

Reply:
The problematic case with `++` in terms of performance is usually appending to an accumulator, since that ends up being quadratic in the number of elements - here you're prepending to an accumulator, so that's why it's not very problematic.